netty/src/main/java/io/grpc/netty/ProtocolNegotiator.java

```diff
@@ -63,4 +65,16 @@ interface ServerFactory {
      */
     ProtocolNegotiator newNegotiator(ObjectPool<? extends Executor> offloadExecutorPool);
   }
+
+  /**
+   * Verify the authority against peer if applicable depending on the transport credential type.
+   * @throws UnsupportedOperationException if the verification should happen but the required
+   *     type of TrustManager could not be found.
+   * @throws SSLPeerUnverifiedException if peer verification failed
+   * @throws CertificateException if certificates have a problem
+   */
+  default boolean mayBeVerifyAuthority(String authority)
+      throws SSLPeerUnverifiedException, CertificateException {
+    return true;
```

Comment:
If we are going to have a default, it should fail.

Reply:
Done.